### ISSUE TITLE
About the test

### ISSUE THREAD

**Issue by zerodohero** [vqdang/hover_net#58]
Hello, I'm trying to map the cell type back into the image as well, but I've encountered an error.
Below, in bold I added the mapping of cell types back to the image code, but after this code was added, I ran compute_stats.py for classification accuracy tests, and the test values were very low, which were normal before this function was added. why??
  ###############insteance  map
        pred_inst = remap_label(pred_inst, by_size=True)
        overlaid_output = visualize_instances(pred_inst, img)
        overlaid_output = cv2.cvtColor(overlaid_output, cv2.COLOR_BGR2RGB)
        cv2.imwrite('%s/%s.png' % (proc_dir, basename), overlaid_output)

        **##################   type  map
        pred_type = remap_label(pred_type, by_size=True)
   
        overlaid_output = visualize_instances(pred_type, img,color=[[255,0,0], [0, 255, 255],[0, 119, 102],[0,255,0], [0,0,255]])
        overlaid_output = cv2.cvtColor(overlaid_output, cv2.COLOR_BGR2RGB)
        cv2.imwrite('%s/%s.png' % (proc_dir_type, basename), overlaid_output)**

   befor  [ 0.74480  0.83085  0.27748  0.53310  0.63373  0.51674]
   after  [ 0.72551  0.10892  0.01070  0.24361  0.00495  0.00000]

**Comment by simongraham** [comment]
This code only regards the overlay and therefore should not effect results that you get. We will need further information to diagnose your problem.

**Comment by zerodohero** [comment]
The reason for this result is because the line remap_label(pred_inst, by_size=True) will re-allocate the number, as long as this line is removed during cell mapping.

**Comment by vqdang** [comment]
@zerodohero first of all, from your snippet
```
pred_type = remap_label(pred_type, by_size=True)
overlaid_output = visualize_instances(pred_type, img,color=[[255,0,0], [0, 255, 255],[0, 119, 102],[0,255,0], [0,0,255]])
```
You are providing very wrong input.

The first argument of `visualize_instance` is the instance map, not the type map.

Secondly, you are applying `remap_label` on the *type _map* where value of each pixel is the nuclei type. `remap_label` is for reordering the instance map such that the objects ID ate contiguous. When you are doing as in the snippet the label will be changed to sthg random.

Please read the source code carefully before making modification.

**Comment by zerodohero** [comment]
yes, I already correct it, but in the next issue( Type mapping), my code is right, but the result is worse.

**Comment by vqdang** [comment]
I will close this issue, your subsequent questuon will be addressed in #59
